src/fs_sys_err.h

```diff
@@ -0,0 +1,23 @@
+/*
+Copyright (C) 2016 The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file LICENSE for details.
+*/
+
+/* These will all be returned in syscalls which have a return type of unit32_t
+ * This means that they will all be very high constants because of two's
+ * complement on low absolute value negatives being nearest to the maximum
+ * value of 2^32. Hopefully this is far out of conflict with bytes_read
+*/
+
+#define ERR_FD_OOR -1   // file descriptor is out of range
```

Comment:
Those would be very likely overlap with bytes read/written as the return value. These negatives being returned in uint32_t types will end up being very high positive values due to two's compliment, anyways.
What would you suggest as the error codes, instead?